File: 백준/Bronze/10811. 바구니 뒤집기/바구니 뒤집기.py
N, M = map(int, input().split())
result = list(range(1, N + 1))

for _ in range(M):
    i, j = map(int, input().split())
    while i < j:
        tmp = result[i - 1]
        result[i - 1] = result[j - 1]
        result[j - 1] = tmp
        i += 1
        j -= 1
        
for r in result:
    print(r, end=" ")
    

File: 백준/Bronze/10811. 바구니 뒤집기/바구니 뒤집기.cs
using System;

class Program {
    static void Main() {
        string[] input = Console.ReadLine().Split();
        int N = int.Parse(input[0]);
        int M = int.Parse(input[1]);

        int[] arr = new int[N];
        for (int x = 0; x < N; ++x)
            arr[x] = x + 1;

        for (int y = 0; y < M; ++y) {
            string[] swapInput = Console.ReadLine().Split();
            int i = int.Parse(swapInput[0]) - 1;
            int j = int.Parse(swapInput[1]) - 1;
            
            while (i < j) {
                int tmp = arr[i];
                arr[i++] = arr[j];
                arr[j--] = tmp;
            }
        }

        Console.WriteLine(string.Join(" ", arr));
    }
}